state/watcher/txnwatcher_test.go

```diff
@@ -291,13 +292,16 @@ func (s *TxnWatcherSuite) TestInsertThenRemove(c *gc.C) {
 func (s *TxnWatcherSuite) TestDoubleUpdate(c *gc.C) {
 	_, hub := s.newWatcher(c, 2)
 
+	hub.setupSync()
```

Comment:
This is the failure point. Syncing after the first change should be the medicine.